src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/EditorConfigNamingStyleParser.NamingRule.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    internal static partial class EditorConfigNamingStyleParser
+    {
+        private static bool TryGeySerializableNamingRule(
+            string namingRuleTitle,
+            SymbolSpecification symbolSpec,
+            NamingStyle namingStyle,
+            IReadOnlyDictionary<string, object> allRawConventions,
+            out SerializableNamingRule serializableNamingRule)
+        {
+            DiagnosticSeverity severity;
+            if (!TryGetRuleSeverity(namingRuleTitle, allRawConventions, out severity))
+            {
+            }
+
+            serializableNamingRule = new SerializableNamingRule()
+            {
+                EnforcementLevel = severity,
+                NamingStyleID = namingStyle.ID,
+                SymbolSpecificationID = symbolSpec.ID
+            };
+            return true;
+        }
+
+        private static bool TryGetRuleSeverity(
+            string namingRuleName,
+            IReadOnlyDictionary<string, object> allRawConventions,
+            out DiagnosticSeverity diagnosticSeverity)
+        {
+            diagnosticSeverity = default(DiagnosticSeverity);
+            var result = (from kvp in allRawConventions
+                          where kvp.Key.Trim() == $"dotnet_naming_rule.{namingRuleName}.severity"
```

Comment:
Isn't the whole point of a dictionary that you can look up the keys faster than iterating over the whole thing?  Why not `Trim` while populating the dictionary, and just use `TryGetValue` here?